### PR TITLE
Refactor Python installation scripts

### PR DESCRIPTION
**Pull request by MaksimZhukov**
In scope of this PR we've reworked `nix_setup_template.sh` and `win_setup_template.ps1` scripts, removed `New-SetupFile` function from `nix-helpers.psm1` and fixed `repositoryName` in `create_release_steps.yml` file

### CODE REVIEW

File: scripts/installers/nix_setup_template.sh [hunk 1]
```diff
@@ -19,8 +19,8 @@ if [ ! -d $PYTHON_TOOLCACHE_PATH ]; then
     mkdir -p $PYTHON_TOOLCACHE_PATH
 fi
 
-echo "Delete Python $MAJOR_VERSION.$MINOR_VERSION.* if installed"
-rm -rf $PYTHON_TOOLCACHE_PATH/$MAJOR_VERSION.$MINOR_VERSION.*
+echo "Delete Python $PYTHON_FULL_VERSION if installed"
+rm -rf $PYTHON_TOOLCACHE_PATH/$PYTHON_FULL_VERSION
```
Comment (maxim-lobanov) [inline scripts/installers/nix_setup_template.sh]: It is the same like `PYTHON_TOOLCACHE_VERSION_PATH`

File: scripts/installers/win_setup_template.ps1 [hunk 2]
```diff
@@ -7,67 +7,87 @@ param(
...
+$PythonArchPath = Join-Path -Path $PythonVersionPath -ChildPath $Architecture
 
-    Get-WmiObject Win32_Product -Filter $filterPython | Foreach-Object {$_.Uninstall() | Out-Null }
+$IsMSI = $PythonExecName -match "msi"
 
-    if (Test-Path -Path "$($VersionIsInstalled.FullName)/${Architecture}") {
-        Write-Host "Python${MajorVersion}.${MinorVersion}/${Architecture} was found in ${PythonToolcachePath}"
+if (-Not (Test-Path $PythonToolcachePath)) {
+    Write-Host "Create Python toolcache folder"
+    New-Item -ItemType Directory -Path $env:AGENT_TOOLSDIRECTORY -Name "Python" | Out-Null
```
Comment (maxim-lobanov) [inline scripts/installers/win_setup_template.ps1]: Please use $PythonToolcachePath

File: scripts/installers/win_setup_template.ps1 [hunk 3]
```diff
@@ -7,67 +7,87 @@ param(
     [String]$PythonExecName = "__PYTHON_EXEC_NAME__"
```
Comment (maxim-lobanov) [inline scripts/installers/win_setup_template.ps1]: Do we really need to pass `PythonExecName` as parameter? If it is just `python` or `python3`, we can determine it dynamically

Comment (vmapetr) [reply scripts/installers/win_setup_template.ps1]: Actually it's not just python or python3. It is the full name of python installer, like `python-3.8.2-amd64.exe` or msi for older pythons.

File: scripts/installers/win_setup_template.ps1 [hunk 4]
```diff
@@ -7,67 +7,87 @@ param(
...
-$MinorVersion = $Version.Minor
+function Get-PythonFilter {
+    param(
+        [Parameter (Mandatory = $true)]
+        [String]$ArchFilter
+    )
 
-$IsMSI = $PythonExecName -match "msi"
+    ### Python 2.7 and 3.4 have no architecture postfix
+    if ((($Version -like '3.4.*') -or $IsMSI) -and $Architecture -eq "x86") {
```
Comment (vmapetr) [inline scripts/installers/win_setup_template.ps1]: Python 3.4 is deprecated, i don't think that we need `($Version -like '3.4.*')` condition. Beside, AFAIK it has msi extension as well.

Comment (MaksimZhukov) [reply scripts/installers/win_setup_template.ps1]: Thanks!

File: scripts/installers/win_setup_template.ps1 (hunk 2 above)
Comment (MaksimZhukov) [reply scripts/installers/win_setup_template.ps1]: Thanks!

File: scripts/installers/nix_setup_template.sh (hunk 1 above)
Comment (MaksimZhukov) [reply scripts/installers/nix_setup_template.sh]: Thanks!

File: scripts/installers/nix_setup_template.sh [hunk 5]
```diff
@@ -1,26 +1,24 @@
+set -e
+
 MAJOR_VERSION={0}
 MINOR_VERSION={1}
 BUILD_VERSION={2}
```
Comment (maxim-lobanov) [inline scripts/installers/nix_setup_template.sh]: I would prefer to use the same replace pattern like Windows `{{__MAJOR_VERSION__}}`

File: scripts/installers/win_setup_template.ps1 [hunk 6]
```diff
@@ -1,73 +1,88 @@
-param(
-    [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()]
-    [String]$Architecture = "__ARCHITECTURE__",
-    [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()]
-    [Version]$Version = "__VERSION__",
-    [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()]
-    [String]$PythonExecName = "__PYTHON_EXEC_NAME__"
-)
+$Architecture = "__ARCHITECTURE__"
```
Comment (maxim-lobanov) [inline scripts/installers/win_setup_template.ps1]: Let's use pattern `{{__VARIABLE__}}`

Comment (MaksimZhukov) [reply scripts/installers/win_setup_template.ps1]: Thanks

**Pull request merged**